### PR TITLE
Let caller of peer.Create handle errors

### LINKED ISSUE #782: Peer should handle the error channel in the Create function
See https://github.com/dusk-network/dusk-blockchain/pull/776#discussion_r530253219

### PR DESCRIPTION
**Pull request by julesdesmit**
Fixes #782

### CODE REVIEW

File: cmd/dusk/server.go [hunk 1]
```diff
@@ -271,7 +273,17 @@ func (s *Server) OnConnection(conn net.Conn, addr string) {
 		log.Panic(err)
 	}
 
-	go peer.Create(context.Background(), peerReader, peerWriter, writeQueueChan)
+	ctx, cancel := context.WithCancel(context.Background())
+	errChan := peer.Create(ctx, peerReader, peerWriter, writeQueueChan)
+	go handlePeerError(errChan, cancel)
```
Comment (autholykos) [inline cmd/dusk/server.go:278]: Returning channels is not very idiomatic in go. IMO, a more readable version would be:

```
func (s *Server) OnAccept(conn net.Conn) {
    ...
    go func() {
        ctx, cancel := context.WithCancel(context.Background())
        defer cancel()
        err := peer.Create(ctx, peerReader, peerWriter, writeQueueChan)
        logServer.WithError(err).Warnln("error received from peer")
    }
    ...
}

func Create(ctx context.Context, reader *Reader, writer *Writer, writeQueueChan <-chan bytes.Buffer) chan error {
        errChan := make(chan error, 1)
        defer close(errChan)

	go reader.ReadLoop(ctx, errChan)
	go writer.Serve(ctx, writeQueueChan, errChan)

        return <-errChan
}
```

**Review by autholykos — CHANGES_REQUESTED** [review]
LGTM, neatpicking on channel returning

File: cmd/dusk/server.go (hunk 1 above)
Comment (julesdesmit) [reply cmd/dusk/server.go]: Just changed it to that :)

**Comment by autholykos** [comment]
LGTM

**Pull request merged**